setup/build_compile.sh

```diff
@@ -0,0 +1,187 @@
+#!/bin/bash
+
+# Build GMP
+function build_gmp() {
```

Comment:
I also don't think it would work as well for the DAGMC build script because not all the packages are built in the same way. (HDF5 from source tar, CUBIT straight from tar, CGM source from repo, MOAB source from repo, DAGMC source from repo but also requires a tar and patch application, FLUKA and Geant4 probably also have non-reusable code...)